extend/flytekit-xgboost/flytekitplugins/xgboost/task.py

```diff
@@ -84,28 +74,65 @@ class XGBoostParameters(Updateable):
         label_column: Index of the column containing the labels.
     """
 
-    model_parameters: ModelParameters = ModelParameters()
-    hyper_parameters: HyperParameters = HyperParameters()
+    model_parameters: Optional[ModelParameters] = None
+    hyper_parameters: Optional[HyperParameters] = None
```

Comment:
It's good if we set the default values here. I don't think there's a necessity to include None at all.